### ISSUE TITLE
An extra newline character gets printed in Python mode

### ISSUE THREAD

**Issue by laloch** [xonsh/xonsh#2910]
## xonfig
```
+------------------+----------------------+
| xonsh            | 0.8.2                |
| Git SHA          | 6159f449             |
| Commit Date      | Sep 21 02:39:43 2018 |
| Python           | 3.7.1                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.4                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```
## Expected Behavior
Prompt should be rendered immediately after the last command's output.

## Current Behavior
PTK2 shell adds an extra newline to the output of commands executed in Python mode.

## Steps to Reproduce
```
laloch@uriel ~ $ echo @(1+1)
2
laloch@uriel ~ $ 1+1
2

laloch@uriel ~ $
```

**Comment by laloch** [comment]
If I understood him right, @scopatz suggested that the extra newline gets there from
https://github.com/xonsh/xonsh/blob/dca4a477150b55a7011078daba9da8bb0e95ceae/xonsh/base_shell.py#L382, but no, unfortunately that's not the case.

**Comment by laloch** [comment]
The extra EOL seems to be printed from within `exec()` call in codecache.py:run_compiled_code(). If I replace the call at [line 67](https://github.com/xonsh/xonsh/blob/f409a020387c6e73a58fbac5191675f0d519591e/xonsh/codecache.py#L67) by
```
    print("<")
    func(code, glb, loc)
    print(">")
```
I get
from `xonsh --shell-type=rl --no-rc`:
```
$ 1+1
<
2
>
$
```
from `xonsh --shell-type=ptk2 --no-rc`:
```
$ 1+1
<
2

>
$
```

**Comment by scopatz** [comment]
Potential fix in #2957

**Comment by laloch** [comment]
Yep, #2957 does the trick. Thanks @scopatz!
